source/features/subducting_plate_models/temperature/mass_conserving.cc

```diff
@@ -0,0 +1,338 @@
+
+          thermal_diffusivity = prm.get<double>("thermal diffusivity");
+          if (thermal_diffusivity < 0)
+            thermal_diffusivity = this->world->thermal_diffusivity;
+
+          adiabatic_heating = prm.get<bool>("adiabatic heating");
+
+          potential_mantle_temperature = this->world->potential_mantle_temperature >= 0
+                                         ?
+                                         this->world->potential_mantle_temperature
+                                         :
+                                         prm.get<double>("potential mantle temperature");
+
+          surface_temperature = this->world->surface_temperature;
+
+        }
+
+
+        double
+        MassConserving::get_temperature(const Point<3> & /*position_in_cartesian_coordinates*/,
+                                        const double depth,
+                                        const double gravity_norm,
+                                        double temperature_,
+                                        const double /*feature_min_depth*/,
+                                        const double /*feature_max_depth*/,
+                                        const WorldBuilder::Utilities::PointDistanceFromCurvedPlanes &distance_from_planes) const
+        {
+
+          const double distance_from_plane = distance_from_planes.distance_from_plane;
+          const double distance_along_plane = distance_from_planes.distance_along_plane;
+          const double depth_slab_surface = distance_from_planes.depth_reference_surface;
+
+          const double seconds_in_year = 60.0*60.0*24.0*365.25;
+          const double plate_age_sec = age_at_trench*1e6*seconds_in_year;  // my --> seconds
+          const double over_plate_age_sec = overriding_plate_age_above_slab*1e6*seconds_in_year;  // my --> seconds
+
+          if (distance_from_plane <= max_depth && distance_from_plane >= min_depth)
+            {
+
+              // 1. Determine initial heat content of the slab based on age of plate at trench
+              //    This uses the integral of the half-space temperature profile
+              const double initial_heat_content = 2*thermal_conductivity*(surface_temperature - potential_mantle_temperature) *
+                                                  std::sqrt(plate_age_sec/(thermal_diffusivity*const_pi));
+
+              //  2. Get Tmin and distance_offset given distance along slab and depth of point on the slab.
+              //  These equations are empirical based on fitting the temperature profiles from dynamic subduction models.
+              double min_temperature = 0.0;
+              double distance_offset = 0.0;
+
+              const double mantle_coupling_length = mantle_coupling_depth/std::sin(shallow_average_dip*const_pi/180.0); //m
+
+              /* Empirical model parameters */
+              double upper_mantle_depth = 660e3; // m
+              double upper_mantle_length = 1000e3; // m
```

Comment:
You do not seem to use these values, so you can just make that into a comment.

Reply:
I do use upper_mantle_length ,but not depth. I have removed the one that is not used.